feat(curriculum): add security and privacy lessons (#68206)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/lecture-understanding-security-and-privacy-in-web-applications/6a3b86eb689999b66965050a.md

---
id: 6a3b86eb689999b66965050a

> title: What are some common security threats and how to mitigate them?

Reviewer: ```suggestion
title: What Are Some Common Security Threats and How to Mitigate Them?
```

To prevent malware, software teams usually run automated updates and patch known software vulnerabilities. Security is a multi-layer thing, so users should also do their part by installing updates immediately and antimalware software on their devices.

**Credential stuffing attacks** target user authentication systems. Attackers use automated tools to test millions of stolen username and password combinations across different websites.

This kind of attack succeeds because many people reuse identical passwords across multiple platforms.

Companies can mitigate this threat by enforcing multi-factor authentication. Multi-factor authentication requires a second verification step, like a mobile authenticator app code, a one-time password (OTP), or email verification.

Even if a hacker steals the password, they cannot access the account without this secondary token.

**Distributed Denial of Service attacks** target system availability. Attackers flood a target website with massive amounts of fake internet traffic that suddenly overloads the server, crashes it, and blocks legitimate customers from accessing services.

> To defend against these attacks, businesses use cloud-based traffic filtering services such as Cloudflare, CloudDNS and DNSFilter.

Reviewer: ```suggestion
To defend against these attacks, businesses use cloud-based traffic filtering services such as Cloudflare, Akamai, and AWS Shield.
```

To counter this internal risk, companies enforce the principle of least privilege. This security standard ensures that workers only access the specific files necessary to complete their daily tasks.

Security teams also monitor system logs to flag unusual data downloads or unauthorized file access.

# --questions--

## --text--

What security principle do companies apply to limit the damage caused by insider threats?

## --answers--

They monitor employee social media activity for suspicious behavior.

### --feedback--

Focus on the standard that ensures workers only see what they need for their daily tasks.

---

They require all employees to sign non-disclosure agreements annually.

### --feedback--

Focus on the standard that ensures workers only see what they need for their daily tasks.

---

They grant full system access to all employees for transparency.

### --feedback--

Focus on the standard that ensures workers only see what they need for their daily tasks.

---

They enforce least privilege to restrict access to necessary files only.

## --video-solution--

4

---

> 

Reviewer: ```suggestion
```

## --text--

What mitigation strategy effectively protects against credential stuffing attacks where stolen passwords are tested across multiple websites?

## --answers--

Enforcing multi-factor authentication as a secondary verification step.

---

Requiring users to change their passwords every thirty days.

### --feedback--

Focus on the additional layer that stops attackers even with a valid password.

---

Blocking all login attempts from foreign IP addresses.

### --feedback--

Focus on the additional layer that stops attackers even with a valid password.

---

Storing passwords in plain text for faster verification.

### --feedback--

Focus on the additional layer that stops attackers even with a valid password.

## --video-solution--

1

---

> 

Reviewer: ```suggestion
```
